server/src/main/scala/services/graphstages/ArrivalsGraphStage.scala

```diff
@@ -260,23 +260,93 @@ class ArrivalsGraphStage(name: String = "",
       mergeBestFieldsFromSources(baseArrival, mergeArrival(key).getOrElse(baseArrival))
     }
 
+
+    def scheduleTimeApproximation(searchKey: UniqueArrival, searchKeyArrivalsSourceType: ArrivalsSourceType, foundArrival: Arrival, arrivalsSourceType: ArrivalsSourceType, logMessage: Boolean = false): Option[UniqueArrival] = {
+
+      val arrivalMap: SortedMap[UniqueArrival, Arrival] = arrivalsSourceType match {
+        case LiveArrivals => liveArrivals //Live feeds arrivals
+        case LiveBaseArrivals => liveBaseArrivals // Cirium feeds arrivals
+        case BaseArrivals => forecastBaseArrivals // ACL feeds arrivals
+      }
+
+      val potentialKeyList: Iterable[UniqueArrival] = arrivalMap.keys.collect {
+        case key if key.number == searchKey.number && key.terminal == searchKey.terminal && Math.abs(key.scheduled - searchKey.scheduled) <= 1 * 60 * 60 * 1000 => key
```

Comment:
Could be nice to move this logic to a function on the UniqueArrival class, with a parameter for number of minutes threshold, and then switch this to a filter